crates/biome_js_analyze/src/aria_analyzers/nursery/no_interactive_element_to_noninteractive_role.rs

```diff
@@ -0,0 +1,98 @@
+    /// Non-interactive ARIA roles include `article`, `banner`, `complementary`, `img`, `listitem`, `main`, `region` and `tooltip`.
+    ///
+    /// Source: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-interactive-element-to-noninteractive-role.md
+    ///
+    /// [WAI-ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#usage_intro)
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <input role="img" />;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```jsx
+    /// <input role="button" />;
+    /// ```
+    ///
+    pub(crate) NoInteractiveElementToNoninteractiveRole {
+        version: "next",
+        name: "noInteractiveElementToNoninteractiveRole",
+        recommended: false,
+    }
+}
+
+impl Rule for NoInteractiveElementToNoninteractiveRole {
+    type Query = Aria<AnyJsxElement>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let aria_roles = ctx.aria_roles();
+        if node.is_element() {
+            let role_attribute = node.find_attribute_by_name("role")?;
+            let role_attribute_static_value = role_attribute.as_static_value()?;
+            let role_attribute_value = role_attribute_static_value.text();
+            let element_name = node.name().ok()?.as_jsx_name()?.value_token().ok()?;
+
+            let attributes = ctx.extract_attributes(&node.attributes());
+            if !aria_roles.is_not_interactive_element(element_name.text_trimmed(), attributes)
+                && !aria_roles.is_role_interactive(role_attribute_value)
+            {
+                // <div> and <span> are considered neither interactive nor non-interactive, depending on the presence or absence of the role attribute.
+                // We don't report <div> and <span> here, because we cannot determine whether they are interactive or non-interactive.
+                let role_sensitive_elements = ["div", "span"];
+                if role_sensitive_elements.contains(&element_name.text_trimmed()) {
+                    return None;
+                }
+
+                // a tag without href is considered non-interactive
+                if element_name.text_trimmed() == "a"
+                    && node.find_attribute_by_name("href").is_none()
+                {
+                    return None;
+                }
+
+                return Some(());
+            }
+        }
+
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Interactive elements should not be assigned non-interactive roles."
+                },
```

Comment:
I think that `markup` is required only when using variables or styles such as `Emphasis`.
```suggestion
                    "Interactive elements should not be assigned non-interactive roles.",
```